docs/tutorials/ruby-driver-create-client.txt

```diff
@@ -555,6 +571,49 @@ When ``#close`` is called on a client by any thread, all connections are closed:
 
     client.close
 
+
+Use with Forking Servers
+------------------------
+
+When using the Mongo Ruby driver with a forking web server such as Unicorn,
+worker processes should generally each have their own Mongo::Client instances.
+This is because:
+
+1. The background threads remain in the parent process and are not transfered
+   to the child process.
+2. File descriptors like network sockets are shared between parent and
+   child processes.
+
+It is easiest to not create any Mongo::Client instances prior to the fork.
+If the parent process needs to perform operations on the MongoDB database,
+reset the client in an ``after_fork`` handler which is defined in
+``unicorn.rb``::
+
+.. code-block:: ruby
+
+    after_fork do |server, worker|
```

Comment:
This ensures that the monitoring threads are restarted after a fork? Is there a solution for the child deadlocking on a mutex acquired in the parent right before the fork?

Reply:
`reconnect` restarts the monitoring thread, but I think you are correct, there are 3 mutexes in cluster and more drastic measures are needed to ensure they are not deadlocked.

Reply:
Reviewing all uses of Mutex I would say that in normal applications they should never deadlock, as the applications shouldn't be forking with any of these mutexes being taken.